### PR TITLE
Add Edit survey page

### PR DESCRIPTION
**Pull request by ismail-syed**
Closes https://github.com/vikrambombhi/SimpleSurvey/issues/87

### Edit survey page

Screen Shot 2020-04-02 at 2 28 14 AM

### After submitting the edited form

Screen Shot 2020-04-02 at 2 25 45 AM

### CODE REVIEW

File: client/src/components/Survey.js [hunk 1]
```diff
@@ -1,89 +1,96 @@
-import React, {useEffect, useState} from "react";
-import {Button, Form, Select, RangeSlider, FormLayout, TextField} from '@shopify/polaris';
+import React, { useEffect, useState } from "react";
```
Comment (ismail-syed) [inline client/src/components/Survey.js:1]: This file is all automated prettier formatting

File: client/src/components/pages/EditSurvey.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  let { id: surveyId } = useParams();
+
+  const [name, setName] = useState("");
+  const [questions, setQuestions] = useState([]);
+
+  // a curry function that we will pass down to EditSurveyForm to update state
+  // This will be our state updated for all our survey files
+  // index will be the index of the question in in survey.questions,
+  // extraOpts is for updating mins/max in range questions or options in option questions
+  function handleQuestionChange(index, extraOpts) {
```
Comment (ismail-syed) [inline client/src/components/pages/EditSurvey.js:17]: @kirinrastogi , The [QuestionList](https://github.com/vikrambombhi/SimpleSurvey/blob/master/client/src/components/QuestionList.js) state management was a good initial stab but it becomes problematic when sharing with other components.

If you're curious about a better pattern to use, this `EditSurvey` and `EditSurveyForm` handles state management quite well. `EditSurvey` is responsible for owning and updating the state. `EditSurveyForm` is a _dumb_ component only responsible for rendering what it's given and updates state with the handlers passed to it. In this case, it's this `handleQuestionChange` function

File: client/src/components/EditSurveyForm.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+        return null;
+      })
+      .filter(question => !!question);
+
+    await Promise.all([updateTitleRequest, questionRequests]);
+    window.location = "/app/admin/surveys";
+  };
+
+  const questionsMarkup = questions.map((question, index) => {
```
Comment (JackyChiu) [inline client/src/components/EditSurveyForm.js:72]: is it worth making a component to share the code with question markup from `Survey.js`?

File: client/src/components/pages/EditSurvey.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  useEffect(() => {
+    async function fetchData() {
+      const res = await fetch(`/api/surveys`);
+      res
+        .json()
+        .then(res => {
+          // eslint-disable-next-line
+          const surveyToEdit = res.find(sur => sur.id == surveyId);
+          console.log("surveyToEdit: ", surveyToEdit);
```
Comment (JackyChiu) [inline client/src/components/pages/EditSurvey.js:50]: do we still need this log?

File: client/src/components/pages/EditSurvey.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  const [name, setName] = useState("");
+  const [questions, setQuestions] = useState([]);
+
+  // a curry function that we will pass down to EditSurveyForm to update state
+  // This will be our state updated for all our survey files
+  // index will be the index of the question in in survey.questions,
+  // extraOpts is for updating mins/max in range questions or options in option questions
+  function handleQuestionChange(index, extraOpts) {
+    return newValue => {
+      const updatedQuestions = questions.map((question, idx) => {
```
Comment (JackyChiu) [inline client/src/components/pages/EditSurvey.js:19]: nit: I find the map misleading for the intention of this function. I feel like changing the value you intend to update is much more clear, something like this
```js
let updatedQuestions = [...questions];
if (extraOpts) {
  updatedQuestions[index] = {
    ...question,
    ...(extraOpts === "min" ? { min: newValue } : {}),
    ...(extraOpts === "max" ? { max: newValue } : {}),
    ...(extraOpts === "options" ? { options: newValue.split(",") } : {})
} else {
  updatedQuestions[index] = {
    ...question,
    question: newValue
  };
}
setQuestions(updatedQuestions); 

```

File: client/src/components/EditSurveyForm.js (hunk 3 above)
Comment (ismail-syed) [reply client/src/components/EditSurveyForm.js:72]: Ideally, yes.

File: client/src/components/pages/EditSurvey.js (hunk 4 above)
Comment (ismail-syed) [reply client/src/components/pages/EditSurvey.js:50]: Nope. it should be removed

**Pull request merged**